src/main/java/com/example/demo/services/UsersService.java

```diff
@@ -125,12 +141,382 @@ public Integer extractData(ResultSet rs) throws SQLException, DataAccessExceptio
+					String subjectID = rs.getString("SUB_ID");
+					String subjectName = getSubjectNameFromSubjectID(subjectID);
+					
+					if(userAcademicDetailsMap.containsKey(academicID))
+					{
+						UserAcademicDetailsBO userAcademicDetails = userAcademicDetailsMap.get(academicID);
+						List<ClassDetaislBO> classDetails = userAcademicDetails.getUserClassSubjectsL();
+						
+						int foundClasID = 0;
+						for(ClassDetaislBO cbo : classDetails)
+						{
+							String classID_list = cbo.getClassId();
+							if(classID_list.equals(classID))
+							{
+								foundClasID=1;
+								SubjectDetailsBO subjectDetailsBO = new SubjectDetailsBO();
+								subjectDetailsBO.setSubjectId(subjectID);
+								subjectDetailsBO.setSubjectName(subjectName);
+								cbo.getSubjects().add(subjectDetailsBO);
+								break;
+							}
+						}
+						if(foundClasID==0)
+						{
+							SubjectDetailsBO subjectDetailsBO = new SubjectDetailsBO();
+							subjectDetailsBO.setSubjectId(subjectID);
+							subjectDetailsBO.setSubjectName(subjectName);
+							List<SubjectDetailsBO> subjects = new ArrayList();
+							subjects.add(subjectDetailsBO);
+							ClassDetaislBO classDet = new ClassDetaislBO();
+							classDet.setClassId(classID);
+							classDet.setClassName(className);
+							classDet.setSubjects(subjects);
+							classDetails.add(classDet);
+						}
+					}
+					else
+					{
+						SubjectDetailsBO subjectDetailsBO = new SubjectDetailsBO();
+						subjectDetailsBO.setSubjectId(subjectID);
+						subjectDetailsBO.setSubjectName(subjectName);
+						
+						List<SubjectDetailsBO> subjects = new ArrayList();
+						subjects.add(subjectDetailsBO);
+						
+						ClassDetaislBO classDetailBO = new ClassDetaislBO();
+						classDetailBO.setClassId(classID);
+						classDetailBO.setClassName(className);
+						classDetailBO.setSubjects(subjects);
+						
+						List<ClassDetaislBO> classDetailsBOList = new ArrayList();
+						classDetailsBOList.add(classDetailBO);
+						
+						UserAcademicDetailsBO userAcademicDetailsBO = new UserAcademicDetailsBO();
+						userAcademicDetailsBO.setUserClassSubjectsL(classDetailsBOList);
+						
+						userAcademicDetailsMap.put(academicID, userAcademicDetailsBO);
+					}
+					
+				}
+				return userAcademicDetailsMap;
+			}
+		});
+	}
+	
+	public List<UserManagerDetailsBO> getManagerDetailsOfOneUser(String userId)
+	{
+		String query = "SELECT * FROM USER_MANAGER_MAPPING where user_id=?";
+		
+		return jdbcTemplate.query(query, new PreparedStatementSetter() {
+			
+			@Override
+			public void setValues(PreparedStatement ps) throws SQLException {
+				ps.setString(1, userId);
+			}
+			
+		}, new ResultSetExtractor<List<UserManagerDetailsBO>>() {
+			
+			@Override
+			public List<UserManagerDetailsBO> extractData(ResultSet rs) throws SQLException, DataAccessException {
+				
+				List<UserManagerDetailsBO> userManagerDetailsBO = new ArrayList(); 
+				while (rs.next()) 
```

Comment:
Need to fetch the name of manager as well . Again write a query in such a way that it will fetch the name in same query

Reply:
need to revisit